driver/image_adjust_hue_driver.hpp

```diff
@@ -0,0 +1,389 @@
+
+        set4DValueAt(output_buf, output_tv, n, 0, h, w, r);
+        set4DValueAt(output_buf, output_tv, n, 1, h, w, g);
+        set4DValueAt(output_buf, output_tv, n, 2, h, w, b);
+    }
+}
+
+template <typename Tgpu, typename Tref>
+class ImageAdjustHueDriver : public Driver
+{
+public:
+    ImageAdjustHueDriver() : Driver()
+    {
+        miopenCreateTensorDescriptor(&inputTensorDesc);
+        miopenCreateTensorDescriptor(&outputTensorDesc);
+
+        data_type = miopen_type<Tgpu>{};
+    }
+
+    int AddCmdLineArgs() override;
+    int ParseCmdLineArgs(int argc, char* argv[]) override;
+    InputFlags& GetInputFlags() override { return inflags; }
+
+    int GetandSetData() override;
+
+    int AllocateBuffersAndCopy() override;
+
+    int RunForwardGPU() override;
+    int RunForwardCPU();
+
+    int RunBackwardGPU() override;
+    int RunBackwardCPU();
+
+    int VerifyForward() override;
+    int VerifyBackward() override;
+
+    ~ImageAdjustHueDriver() override
+    {
+        miopenDestroyTensorDescriptor(inputTensorDesc);
+        miopenDestroyTensorDescriptor(outputTensorDesc);
+    }
+
+private:
+    InputFlags inflags;
+
+    int forw;
+
+    miopenTensorDescriptor_t inputTensorDesc;
+    miopenTensorDescriptor_t outputTensorDesc;
+
+    std::unique_ptr<GPUMem> input_gpu;
+    std::unique_ptr<GPUMem> output_gpu;
+
+    std::vector<Tgpu> in_host;
+    std::vector<Tgpu> out_host;
+
+    std::vector<Tref> out_ref;
+
+    float hue;
+};
+
+template <typename Tgpu, typename Tref>
+int ImageAdjustHueDriver<Tgpu, Tref>::AddCmdLineArgs()
+{
+    inflags.AddInputFlag("forw", 'F', "1", "Run only the forward pass", "int");
+
+    inflags.AddTensorFlag("input", 'I', "1x3x96x96", "Input Tensor Size");
+    inflags.AddInputFlag("contiguous", 'Z', "1", "Use Contiguous Tensors", "int");
+    inflags.AddInputFlag("hue", 'H', "0.2", "Hue", "double");
+
+    inflags.AddInputFlag("iter", 'i', "10", "Number of iterations", "int");
+    inflags.AddInputFlag("verify", 'V', "1", "Verify Each Layer (Default=1)", "int");
+    inflags.AddInputFlag("time", 't', "0", "Time Each Layer (Default=0)", "int");
+    inflags.AddInputFlag(
+        "wall", 'w', "0", "Wall-clock Time Each Layer, Requires time == 1 (Default=0)", "int");
+
+    return miopenStatusSuccess;
+}
+
+template <typename Tgpu, typename Tref>
+int ImageAdjustHueDriver<Tgpu, Tref>::ParseCmdLineArgs(int argc, char* argv[])
+{
+    inflags.Parse(argc, argv);
+
+    hue = inflags.GetValueDouble("hue");
+
+    if(inflags.GetValueInt("time") == 1)
+    {
+        miopenEnableProfiling(GetHandle(), true);
+    }
+    return miopenStatusSuccess;
+}
+
+template <typename Tgpu, typename Tref>
+int ImageAdjustHueDriver<Tgpu, Tref>::GetandSetData()
+{
+    TensorParameters input_vec = inflags.GetValueTensor("input");
+    assert(input_vec.lengths.size() == 4 || input_vec.lengths.size() == 3);
+    if(input_vec.lengths.size() == 3)
+    {
+        // n=1
```

Comment:
Please remove comment

Reply:
Changed the comment to better reflect why an insertion of `n=1` is necessary